### PR TITLE
Add optional proxyLocation variable for containers running as non-root

### PR DESCRIPTION
**Pull request by holzman**
Some kubernetes deployments (i.e. OpenShift) prefer to run containers
as ephemeral users. This optional variable allows the user to set the
x509 proxy location to somewhere writable by non-root (e.g. /tmp)

### CODE REVIEW

**Comment by BenGalewsky** [comment]
Thank you for this, @holzman - It makes it harder to understand the system if there is a default in the docker image that can be overridden here. I would prefer that you move the default to `values.yaml` and always pick it up from the Environment var.

**Comment by holzman** [comment]
Sure, I can do that. (I didn't do it originally since it will break things slightly on upgrade since it adds two new items to values.yaml..)

File: servicex/values.yaml [hunk 1]
```diff
@@ -71,6 +71,7 @@ didFinder:
   rucio_host: https://voatlasrucio-server-prod.cern.ch:443
   auth_host: https://voatlasrucio-auth-prod.cern.ch:443
   threads: 5
+  proxyLocation: /tmp/x509up
```
Comment (BenGalewsky) [inline servicex/values.yaml:74]: Should the default be `/etc/grid-security/x509up` to match existing deployments?

Comment (holzman) [reply servicex/values.yaml:74]: My thought had been to provide a default that worked out of the box for both non-root and root containers, but as long as it's configurable I don't really care.

**Comment by hollowec** [comment]
Another option to solve this problem is to make /etc/grid-security group root writable in the container images, since Openshift/OKD containers run with group root by default.  This is what was done on the ServiceX deployment on OKD at BNL.

**Comment by holzman** [comment]
That's another approach that will work.  My intent was to make things look as familiar as possible (/etc/grid-security not normally writable, proxies live in /tmp)  but I'm OK with just making /etc/grid-security 770.

**Comment by BenGalewsky** [comment]
@holzman Your last comment makes me think we really should bite the bullet and just change all of our images to put the writable proxies in /tmp if that's a normal way of doing things - I'll change everything to make this the hardcoded way and keep config simple.

I appreciate this attempt to not interfere with existing code, but I think this will be better.
Yes?

**Comment by holzman** [comment]
Sounds good to me.

**Comment by hollowec** [comment]
Works for me also.  Thanks!

**Comment by BenGalewsky** [comment]
This PR is obsolete now that we've hardcoded placing user certs in `/tmp` in all of the impacted containers

**Pull request closed without merging**